autoload/vsnip.vimspec

```diff
@@ -0,0 +1,98 @@
+let s:expect = themis#helper('expect')
+
+Describe vsnip
+
+  Describe #get_context
+
+    It should return context information in insert-mode
+      function! Test()
+        call s:expect(mode(1)).to_equal('i')
+        call s:expect(vsnip#get_context()).to_equal({
+        \   'range': {
+        \     'start': {
+        \       'line': 0,
+        \       'character': 0,
+        \     },
+        \     'end': {
+        \       'line': 0,
+        \       'character': 5,
+        \     },
+        \   },
+        \   'snippet': {
+        \     'label': 'Test1',
+        \     'description': '',
+        \     'prefix': ['test1'],
+        \     'prefix_alias': [],
+        \     'body': [
+        \       'snippet'
+        \     ],
+        \   }
+        \ })
+      endfunction
+      enew!
+      call setline(1, 'test1')
+      call cursor([1, 5])
+      call feedkeys("a\<C-r>=Test()\<CR>", 'x')
+    End
+
+    It should return context information in normal-mode
+      enew!
+      call setline(1, 'test1')
+      call cursor([1, 5])
+      call s:expect(mode(1)).to_equal('n')
+      call s:expect(vsnip#get_context()).to_equal({
+      \   'range': {
+      \     'start': {
+      \       'line': 0,
+      \       'character': 0,
+      \     },
+      \     'end': {
+      \       'line': 0,
+      \       'character': 5,
+      \     },
+      \   },
+      \   'snippet': {
+      \     'label': 'Test1',
+      \     'description': '',
+      \     'prefix': ['test1'],
+      \     'prefix_alias': [],
+      \     'body': [
+      \       'snippet'
+      \     ],
+      \   }
+      \ })
+    End
+
+    It should return context information in select-mode
+      enew!
+      call setline(1, 'test1')
+      call feedkeys("v\<C-g>", 'x')
```

Comment:
I recently found the `x` flag for `feedkeys`.
It is super useful to write a test!